### DISCUSSION TITLE
False positive detection

### DISCUSSION THREAD

**Discussion by arudinskis** [DependencyTrack/dependency-track#1850] [General]
DT report that Ruby on Rails version 7.0.3.1 component `activerecord` has 2 high severity vulnerabilities:
- CVE-2017-17916 (NVD): SQL injection vulnerability in the 'find_by' method in Ruby on Rails 5.1.4 and earlier allows remote attackers to execute arbitrary SQL commands via the 'name' parameter. NOTE: The vendor disputes this issue because the documentation states that this method is not intended for use with untrusted input.
- CVE-2017-17917 (NVD): SQL injection vulnerability in the 'where' method in Ruby on Rails 5.1.4 and earlier allows remote attackers to execute arbitrary SQL commands via the 'id' parameter. NOTE: The vendor disputes this issue because the documentation states that this method is not intended for use with untrusted input.

This component CPE `cpe:2.3:a:activerecord:activerecord:7.0.3.1:*:*:*:*:*:*:*` clearly define component name and version. Both vulnerabilities are related with Rails < 5.1.4 and do not apply to this component.

Why do DT analyzers report such findings?

**Comment by officerNordberg** [comment]
Which analyzer? I'm assuming you are on a release and not a development
version. I can try to take a look at this next week. CPE matching
definitely gets whacky but sadly there is usually a NVD data related
explanation for the insanity but I don't see that so far here.
